src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest.kt

```diff
@@ -123,46 +123,92 @@ class OffenderResourceIntTest : ResourceTest() {
+        OFFENDER_NUMBER,
+      )
+      assertThatJsonFileAndStatus(response, 200, "casenotes.json")
+    }
 
-  @Test
-  fun testViewCaseNotesRoleCanRetrieveCaseNotesForOffender() {
-    val token = authTokenHelper.getToken(AuthToken.VIEW_CASE_NOTES)
-    val httpEntity = createHttpEntity(token, null)
-    val response = testRestTemplate.exchange(
-      "/api/offenders/{nomsId}/case-notes/v2",
-      GET,
-      httpEntity,
-      object : ParameterizedTypeReference<String?>() {},
-      OFFENDER_NUMBER,
-    )
-    assertThatJsonFileAndStatus(response, 200, "casenotes.json")
-  }
+    @Test
+    fun testViewCaseNotesRoleCanRetrieveCaseNotesForOffender() {
+      val token = authTokenHelper.getToken(AuthToken.VIEW_CASE_NOTES)
+      val httpEntity = createHttpEntity(token, null)
+      val response = testRestTemplate.exchange(
+        "/api/offenders/{nomsId}/case-notes/v2",
+        GET,
+        httpEntity,
+        object : ParameterizedTypeReference<String?>() {},
+        OFFENDER_NUMBER,
+      )
+      assertThatJsonFileAndStatus(response, 200, "casenotes.json")
+    }
 
-  @Test
-  fun testCannotRetrieveCaseNotesForOffenderWithViewPrisonerData() {
-    val token = authTokenHelper.getToken(VIEW_PRISONER_DATA)
-    val httpEntity = createHttpEntity(token, null)
-    val response = testRestTemplate.exchange(
-      "/api/offenders/{nomsId}/case-notes/v2",
-      GET,
-      httpEntity,
-      object : ParameterizedTypeReference<String?>() {},
-      OFFENDER_NUMBER,
-    )
-    assertThat(response.statusCode.value()).isEqualTo(403)
+    @Test
+    fun testCanFilterCaseNotesByType() {
+      val token = authTokenHelper.getToken(AuthToken.NORMAL_USER)
+      val httpEntity = createHttpEntity(token, null)
+      val response = testRestTemplate.exchange(
+        "/api/offenders/{nomsId}/case-notes/v2?type=ETE",
+        GET,
+        httpEntity,
+        object : ParameterizedTypeReference<String?>() {},
+        OFFENDER_NUMBER,
+      )
+      assertThatJsonFileAndStatus(response, 200, "casenotes_type_ETE.json")
+    }
+
+    @Test
+    fun testCanFilterCaseNotesByTypeAndSubType() {
+      val token = authTokenHelper.getToken(AuthToken.NORMAL_USER)
+      val httpEntity = createHttpEntity(token, null)
+      val response = testRestTemplate.exchange(
+        "/api/offenders/{nomsId}/case-notes/v2?type=ETE&subType=ETERTO",
+        GET,
+        httpEntity,
+        object : ParameterizedTypeReference<String?>() {},
+        OFFENDER_NUMBER,
+      )
+      assertThatJsonFileAndStatus(response, 200, "casenotes_type_ETE.json")
+    }
+
+    @Test
+    fun testCanFilterCaseNotesByMultipleTypeAndSubTypes() {
+      val token = authTokenHelper.getToken(AuthToken.NORMAL_USER)
+      val httpEntity = createHttpEntity(token, null)
+      val response = testRestTemplate.exchange(
+        "/api/offenders/{nomsId}/case-notes/v2?type=ETE&subType=ETERTO&caseNoteTypeSubTypes=APP&caseNoteTypeSubTypes=COMMS+COM_IN",
```

Comment:
I think this should be a `400` error instead.  The user has asked for type `ETE` but then also asked to filter by additional types too

Reply:
Added validation and test to return `400` when type and typeSubTypes are set.